specs/online_order_spec.rb

```diff
@@ -11,66 +13,104 @@
+      online_order_test.must_be_kind_of(Grocery::Order)
     end
 
     it "Can access Customer object" do
       # TODO: Your test code here!
+      all_online_orders = Onlineorder.all
+      online_order_test = all_online_orders[0]
+      online_order_test.customer.must_be_instance_of(Grocery::Customer)
     end
 
     it "Can access the online order status" do
       # TODO: Your test code here!
+      all_online_orders = Onlineorder.all
+      online_order_test = all_online_orders[0]
+      online_order_test.status.must_equal(:complete)
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      # # TODO: Your test code here!
+      all_online_orders = Onlineorder.all
+      online_order_test = all_online_orders[0]
+      online_order_test.total.must_be_within_delta 180.67, 0.1
+
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      an_order = Onlineorder.new('4',nil,nil,nil,nil)
+      an_order.total.must_equal(0)
+
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      #completed
+      online_orders = Onlineorder.all
+      proc {online_orders.add_product('apple',4.00,:complete)}.must_raise 'Input is not pending or paid'
+
+      #shipped
+      online_orders = Onlineorder.all
+      proc {online_orders.add_product('apple',4.00,:shipped)}.must_raise 'Input is not pending or paid'
     end
 
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+      # # TODO: Your test code here!
+      # online_orders = Onlineorder.all
+      #
+      # online_orders.add_product('very expensive apple for testing',400.00,:pending)
+      # online_orders[online_orders.length-1].products.must_equal 'very expensive apple for testing'
+
     end
   end
 
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
       # TODO: Your test code here!
+      online_orders = Onlineorder.all
+      online_orders.must_be_kind_of Array
+
     end
 
     it "Returns accurate information about the first online order" do
       # TODO: Your test code here!
+      online_orders = Onlineorder.all
+      online_orders[0].id.must_equal "1"
+      online_orders[0].status.must_equal :complete
     end
 
     it "Returns accurate information about the last online order" do
       # TODO: Your test code here!
+      online_orders = Onlineorder.all
+      online_orders[online_orders.length-1].id.must_equal "100"
+      online_orders[online_orders.length-1].status.must_equal :pending
     end
   end
 
   describe "OnlineOrder.find" do
     it "Will find an online order from the CSV" do
       # TODO: Your test code here!
+      online_orders = Onlineorder.all
+      searched = online_orders.find('45')
+      searched.id.must_equal '45'
```

Comment:
You get an error when running this test-- this test isn't written correctly. Why though?

the error is `searched` doesn't have a method `.id`. What is the value of `searched`?

The value of `searched` isn't an instance of `OnlineOrder`, which is what we expected. It's an instance of `Enumerator`. Why?

You set `searched` to the value of `online_orders.find('45')`. `online_orders` is an array of `OnlineOrder` instances. Therefore, Ruby calls on Array's implementation of `.find`.

If you want to use the `.find` method you defined as a class method off of `OnlineOrder`, you're going to need to call it with the correct syntax: `Grocery::OnlineOrder.find('45')`